doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,851 @@
+    }
+    PackageMatchFilter CreatePackageMatchFilter() {
+        return ActivateByCoCreate<PackageMatchFilter>(CLSID_PackageMatchFilter);
+    }
+```
+
+## 3.1. Search
+
+The api can be used to search for packages in a catalog known to Windows Package Manager. This can
+be used to get availability information or start an install.
+
+```c++ (C++ish pseudocode)
+
+    IAsyncOperation<CatalogPackage> FindPackageInCatalog(AppCatalog catalog, std::wstring packageId)
+    {
+        FindPackagesOptions findPackagesOptions = CreateFindPackagesOptions();
+        PackageMatchFilter filter = CreatePackageMatchFilter();
+        filter.IsAdditive(true);
+        filter.Field(PackageMatchField::Id);
+        filter.Type(MatchType::Exact);
+        filter.Value(packageId);
+        findPackagesOptions.Filters().Append(filter);
+        FindPackagesResult findPackagesResult{ co_await catalog.FindPackagesAsync(findPackagesOptions) };
+
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        co_return matches.GetAt(0).CatalogPackage();
+    }
+
+    IAsyncOperation<CatalogPackage> MainPage::FindPackage()
+    {
+        // Capture the ui thread context.
+        co_await winrt::resume_background();
+
+        AppInstaller appInstaller = CreateAppInstaller();
+        AppCatalog catalog{ appInstaller.GetAppCatalog(PredefinedAppCatalog::OpenWindowsCatalog) };
+        catalog.OpenAsync().get();
+        co_return FindPackageInCatalog(catalog, m_installAppId).get();
+    }
+```
+
+## 3.2. Install
+
+```c++ (C++ish pseudocode)
+
+    IAsyncOperationWithProgress<InstallResult, InstallProgress> InstallPackage(CatalogPackage package)
+    {
+        AppInstaller appInstaller = CreateAppInstaller();
+        InstallOptions installOptions = CreateInstallOptions();
+
+        installOptions.AppInstallScope(AppInstallScope::User);
+        installOptions.CatalogPackage(package);
+        installOptions.AppInstallMode(AppInstallMode::Silent);
+
+        return appInstaller.InstallPackageAsync(installOptions);
+    }
+
+    IAsyncAction UpdateUIProgress(
+        InstallProgress progress,
+        winrt::Windows::UI::Xaml::Controls::ProgressBar progressBar,
+        winrt::Windows::UI::Xaml::Controls::TextBlock statusText)
+    {
+        co_await winrt::resume_foreground(progressBar.Dispatcher());
+        progressBar.Value(progress.DownloadPercentage);
+
+        std::wstring downloadText{ L"Downloading. " };
+        switch (progress.State)
+        {
+        case AppInstallProgressState::Queued:
+            statusText.Text(L"Queued");
+            break;
+        case AppInstallProgressState::Downloading:
+            downloadText += std::to_wstring(progress.BytesDownloaded) + L" bytes of " + std::to_wstring(progress.BytesRequired);
+            statusText.Text(downloadText.c_str());
+            break;
+        case AppInstallProgressState::Installing:
+            statusText.Text(L"Installing");

```

Comment:
For the Installing state, how can the callback differentiate between determinate and indeterminate progress?

Reply:
In the Downloading state DownloadPercentage and BytesDownloaded/BytesReceived are updated in order to show determinate progress. In the Installing state there's nothing like that and only indeterminate progress is possible.

Reply:
I can update the sample for this, I think you'd just add 
     `   progressBar.IsIndeterminate(true); `
right under this.

Reply:
For msix we will have an accurate installation progress, right? Can this API be extended so that we can have install progress for products that provide it?